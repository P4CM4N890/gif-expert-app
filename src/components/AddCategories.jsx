import { useState } from 'react'

export const AddCategories = ( { onNewCategory } ) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = ( { target } ) => {
        setInputValue( target.value );
    }

    const onSubmit = ( event ) => {
        event.preventDefault();
        const value = inputValue.trim();
        if( value.length <= 1 ) return;

        onNewCategory( value );
        setInputValue('');
    }

    return (
            <form onSubmit={ onSubmit } className='browser'>
                <input type='text'
                    placeholder='Buscar gifs'
                    value={ inputValue }
                    onChange={ onInputChange }
                />
                <button onClick={ () => {} }>Agregar</button>
            </form>
    )
}
