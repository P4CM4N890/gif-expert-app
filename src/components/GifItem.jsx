import PropTypes from 'prop-types';

export const GifItem = ( { title, url } ) => {

    return (
        <div className="card">
            <img src={ url } alt={ title } />
            <p>{ ( title.length > 1 ) ? title : 'Sin t√≠tulo' }</p>
        </div>
    );
}

GifItem.propTypes = {
    title   : PropTypes.string.isRequired,
    url     : PropTypes.string.isRequired,
}