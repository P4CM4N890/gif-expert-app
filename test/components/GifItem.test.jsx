import { render } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en GifItem', () => {

    const title = 'One Punch Man GIF';
    const url   = 'https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=9179ae8etwtm652xi57dbevkyr2vd6j4xka0mx0zeghl7fo1&ep=v1_gifs_search&rid=giphy.gif&ct=g';
    
    test('El componente debe ser igual a la snapshot', () => {
        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();
    });

});