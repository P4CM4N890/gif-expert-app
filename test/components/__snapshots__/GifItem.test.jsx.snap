// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem El componente debe ser igual a la snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man GIF"
      src="https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=9179ae8etwtm652xi57dbevkyr2vd6j4xka0mx0zeghl7fo1&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Man GIF
    </p>
  </div>
</div>
`;
